### PR TITLE
Refactor core

### PR DESCRIPTION
**Pull request by VoIdJMp**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { version } from '../package.json';
 lazyload.version = version;
 lazyload.install = function (Vue) {
     // eslint-disable-next-line no-undef
-    Vue.directive(Slider.name, Slider);
+    Vue.component(Slider.name, Slider);
```
Comment (fayzzzm) [inline src/index.js:7]: From where do you get this `Slider`, still I don't know... Could you explain to me this part of your code?

File: src/lazyloading.js [hunk 2]
```diff
@@ -1,53 +1,136 @@
...
+        this.firstStepToLoad = false;
+    },
+    /**
+     * Event on load image
+     * @returns {null}
+     * */
+    onLoadImage() {
+      if (!this.firstStepToLoad) {
+          this.changeElementAttribute(['class', 'style'], [this.image.class, '']);
+      }
```
Comment (fayzzzm) [inline src/lazyloading.js:100]: You don't need this line of your code. Through, you will firstly download preview, which means, you will get a preview. Or you think there will be some moment, where the preview won't download first and the image will be?

File: src/lazyloading.js [hunk 3]
```diff
@@ -1,53 +1,136 @@
...
+    options: {
+      type: Object,
+      default: () => {
+        return observerDefault
+      },
+    },
+  },
+  data() {
+    return {
+      observer: {},
```
Comment (fayzzzm) [inline src/lazyloading.js:41]: Where do you use it?

File: src/lazyloading.js [hunk 4]
```diff
@@ -1,53 +1,136 @@
...
+      firstStepToLoad: true,
+      image: {
+        class: '',
+        src: '',
+      },
+    }
+  },
+  render() {
+      return this.$slots.default;
+  },
```
Comment (fayzzzm) [inline src/lazyloading.js:51]: Put it at the end of your code.

File: src/lazyloading.js [hunk 5]
```diff
@@ -1,53 +1,136 @@
...
 
-        lazyImage();
+          this.changeElementAttribute(
+              ['class', 'style', 'src'],
+              [
+                  this.placeholderImage.class,
+                  this.placeholderImage.style,
+                  this.placeholderImage.src,
+              ]
+          );
```
Comment (fayzzzm) [inline src/lazyloading.js:83]: Looks weird at least. Could be done in a more efficient way. Why can we put the placeholder first and then change it to the original?

File: src/lazyloading.js [hunk 6]
```diff
@@ -1,53 +1,136 @@
...
+      }
+    },
+    /**
+     * Event on error image
+     * @returns {null}
+     * */
+    onErrorImage() {
+        this.changeElementAttribute(
+            ['class', 'style', 'src'],
+            [this.image.class, '', require('./assets/img/photo.png')]
```
Comment (fayzzzm) [inline src/lazyloading.js:109]: `onErrorImage` - the image should be given as props. The user decides what picture should be given to the website, not we...

File: src/index.js (hunk 1 above)
Comment (VoIdJMp) [reply src/index.js:7]: OMG!!!

Comment (VoIdJMp) [reply src/index.js:7]: there should be a lazyload

Comment (VoIdJMp) [reply src/index.js:7]: pls change it

Comment (VoIdJMp) [reply src/index.js:7]: Thanks :)

File: src/lazyloading.js (hunk 2 above)
Comment (VoIdJMp) [reply src/lazyloading.js:100]: No, bro there is a moment where I change this.firstStepToLoad to false

File: src/lazyloading.js (hunk 3 above)
Comment (VoIdJMp) [reply src/lazyloading.js:41]: image
image

File: src/lazyloading.js (hunk 4 above)
Comment (VoIdJMp) [reply src/lazyloading.js:51]: Ok bro thanks
It's my wrong

File: src/lazyloading.js (hunk 5 above)
Comment (VoIdJMp) [reply src/lazyloading.js:83]: I thought about it too. I can create two templates

File: src/lazyloading.js (hunk 6 above)
Comment (VoIdJMp) [reply src/lazyloading.js:109]: ok bro I'll fix it

File: src/lazyloading.js (hunk 2 above)
Comment (VoIdJMp) [reply src/lazyloading.js:100]: image

File: src/lazyloading.js (hunk 3 above)
Comment (fayzzzm) [reply src/lazyloading.js:41]: But you are not calling it as `this.observer`. Look at your code, each time you are initializing a new variable with this name and you haven't used this parameter from your data object.

File: src/lazyloading.js (hunk 2 above)
Comment (fayzzzm) [reply src/lazyloading.js:100]: Got it!

**Pull request merged**